From f6464ce9d2be36135b25dba01ef77b38d3b84f87 Mon Sep 17 00:00:00 2001
From: yingzhex <yingzhenx.li@intel.com>
Date: Thu, 3 Dec 2020 18:04:33 +0800
Subject: [PATCH] Create application cache directories through File.

Try to recreate application cache directories through File
if we failed to create application cache directories
through StorageManager due to permission issue.

Change-Id: If40500aa394ce021c868f49df91949c5b2723bbf
Tracked-On: ACP-1326
Signed-off-by: yingzhex <yingzhenx.li@intel.com>
---
 core/java/android/app/ContextImpl.java | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/core/java/android/app/ContextImpl.java b/core/java/android/app/ContextImpl.java
index 505b498e3cf..2eb1db3e307 100644
--- a/core/java/android/app/ContextImpl.java
+++ b/core/java/android/app/ContextImpl.java
@@ -2883,8 +2883,10 @@ class ContextImpl extends Context {
                         }
                     }
                 } catch (Exception e) {
-                    Log.w(TAG, "Failed to ensure " + dir + ": " + e);
-                    dir = null;
+                    if(!dir.mkdirs()) {
+                        Log.w(TAG, "Failed to ensure " + dir + ": " + e);
+                        dir = null;
+                    }
                 }
             }
             result[i] = dir;
-- 
2.25.1

